File: src/main/java/net/zenxarch/bot/command/MineCommand.java
package net.zenxarch.bot.command;

import static net.fabricmc.fabric.api.client.command.v1.ClientCommandManager.*;

import com.mojang.brigadier.CommandDispatcher;
import dev.xpple.clientarguments.arguments.CBlockStateArgumentType;
import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource;
import net.zenxarch.bot.process.MineProcess;

public class MineCommand {
  public static void register(
      CommandDispatcher<FabricClientCommandSource> dispatcher) {
    dispatcher.register(literal("zmine").then(
        argument("block", CBlockStateArgumentType.blockState())
            .executes(ctx -> {
              MineProcess.mineBlock(
                  CBlockStateArgumentType.getCBlockState(ctx, "block")
                      .getBlock());
              return 0;
            })));
  }
}

File: src/main/scala/net/zenxarch/bot/command/MineCommand.scala
package net.zenxarch.bot.command

import net.fabricmc.fabric.api.client.command.v1.ClientCommandManager._

import com.mojang.brigadier.CommandDispatcher
import dev.xpple.clientarguments.arguments.CBlockStateArgumentType._
import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource
import net.zenxarch.bot.process.MineProcess

class MineCommand extends ZenCommand {
  def register() = literal("zmine").`then`(
    argument("block", blockState())
      .executes((ctx) => {
        MineProcess.mineBlock(
          getCBlockState(ctx, "block")
            .getBlock()
        )
        0
      })
  )
}
